src/Agent.Listener/Configuration.FreeBSD/DaemonControlManager.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(DaemonControlManager))]
+    public interface IFreeBSDServiceControlManager : IAgentService
+    {
+        void GenerateScripts(AgentSettings settings);
+    }
+
+    public class DaemonControlManager : ServiceControlManager, IFreeBSDServiceControlManager
```

Comment:
Just a nitpick. The common term on FreeBSD is `service` or `rc service` for these type of things.

Maybe replace all occurencies of `Daemon` word with `RCService` or `RunComService`?

Reply:
_Service_ is a general term that applies to almost all operating systems, but I think the word _Daemon_ fits FreeBSD well in this case, as it matches the description in chapter [3.8 Processes and Daemons](https://docs.freebsd.org/en/books/handbook/basics/#basics-processes) of the FreeBSD handbook quite well, while _rc service_ and _com service_ don't appear there:

> Some programs are not designed to be run with continuous user input and disconnect from the terminal at the first opportunity. For example, a web server responds to web requests, rather than user input. Mail servers are another example of this type of application. These types of programs are known as _daemons_.

Also, `/usr/bin/daemon` is used in [vsts.agent.daemon.template](https://github.com/joperator/azure-pipelines-agent/blob/bb8c3c6ab7da22a46fd8170186e90325dbc11dd5/src/Misc/layoutbin/vsts.agent.daemon.template#L20) to run `./runsvc.sh`, which starts the Azure Pipelines Agent.
The suffix _daemon_ is also displayed in the command of the process in process viewers such as htop.